src/util.rs

```diff
@@ -170,4 +185,18 @@ mod tests {
         assert_eq!(Vec::<PathBuf>::new(), get_expired_files(&current_dir));
         Ok(())
     }
+
+    #[test]
+    fn test_safe_join_path() {
+        assert!(safe_path_join("/foo", "bar").is_some());
+        assert_eq!(
+            safe_path_join("/foo/bar", "baz/"),
+            Some("/foo/bar/baz/".into())
+        );
+
+        assert!(safe_path_join("/foo", "/foobar").is_none());
+        assert!(safe_path_join("/foo", "/bar").is_none());
+        assert!(safe_path_join("/foo/bar", "..").is_none());
+        assert!(safe_path_join("/foo/bar", "../").is_none());
```

Comment:
Can you extend the tests for `/foo/../../` + `bar` and different variants of that?